src/rws_interface.cpp

```diff
@@ -818,6 +818,19 @@ bool RWSInterface::setMotorsOff()
   return rws_client_.setMotorsOff().success;
 }
 
+bool RWSInterface::setSpeedRatio(unsigned int ratio)
+{
+  if(ratio > 100)
+  {
+    ratio = 100;
+  }
```

Comment:
I would maybe even go so far as to simply throw an exception on values outside the allowed range.

I like to fail early and fast. And be loud about it.

If the precondition is `0 <= ratio <= 100`, then callers should simply not pass something outside that range. If they do, things will complain, and it's their responsibility to handle that.

Clamping is nice, but in the end does users a disservice I believe: it doesn't alert them to the fact they're providing invalid values.

Reply:
Added throwing of exceptions in 4a99d0a.

> Clamping is nice, but in the end does users a disservice I believe: it doesn't alert them to the fact they're providing invalid values.

I see your point, it would be a [björntjänst](https://sv.wikipedia.org/wiki/Bj%C3%B6rntj%C3%A4nst) as we would say in Sweden 😉.